simulator/func_sim/rf/rf.h

```diff
@@ -31,7 +32,7 @@ class RF
     auto read( Register num) const
     {
         assert( !num.is_mips_hi_lo());
-        return get_entry( num).value;
+        return static_cast<uint32>(get_entry( num).value);
     }
 
     void write( Register num, uint64 val)
```

Comment:
There are too many possibilities to have here:
* uint64 for MIPS32
* uint128 for MIPS64
* uint32 for RISCV32
* uint64 for RISCV64
* uint128 for RISCV128

Let's try to use template here:
```c++
template<typename T>
void write( Register num, const T& val)
```